### PR TITLE
Create generic codemod to replace custom properties

### PR DESCRIPTION
**Pull request by aaronccasanova**
This PR introduces a generic codemod to replace CSS custom properties with mapped values (either a different Polaris custom property or a static value).

```diff
- color: var(--p-text);
+ color: var(--p-color-text);

- animation-duration: var(--p-fast);
+ animation-duration: 100ms;
```

<br />

The migration can be run in two ways:

1. Using a combination of `--decl`, `--from`, and `--to` flags to replace a single custom property, or
2. Using a `--maps` flag which points to a file defining replacements for multiple custom properties

#### Option 1:

```sh
npx @shopify/polaris-migrator styles-replace-custom-property \
  --decl="<name>" --from="<prop>" --to="<prop>" <path>
```

#### Option 2:

```sh
npx @shopify/polaris-migrator styles-replace-custom-property \
  --maps="<replacement-maps>" <path>
```

Example `replacement-maps.js` (with option 2):

```js
export default {
  color: {
    '--p-text': '--p-color-text',
  },
  '/^animation/': {
    '--p-fast': '100ms',
  },
};
```

### CODE REVIEW

File: polaris-migrator/src/migrations/styles-replace-custom-property/tests/replacement-maps-basic.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const colorMap = {
+  '--p-text': '--p-color-text',
+  '--p-black': '#000',
+};
+
+// eslint-disable-next-line import/no-anonymous-default-export, import/no-default-export
+export default {
+  color: colorMap,
+};
```
Comment (alex-page) [inline polaris-migrator/src/migrations/styles-replace-custom-property/tests/replacement-maps-basic.js]: Could we merge this and the regex maps? I don't see a lot of value in separating them.

**Review by alex-page — APPROVED** [review]
Looks good, one non blocking comment.

**Review by sam-b-rose — APPROVED** [review]
This is great! Nice work creating this generic codemod 💯 the two options for usage are nice as well

File: polaris-migrator/src/migrations/styles-replace-custom-property/tests/replacement-maps-basic.js (hunk 1 above)
Comment (aaronccasanova) [reply polaris-migrator/src/migrations/styles-replace-custom-property/tests/replacement-maps-basic.js]: Good catch! https://github.com/Shopify/polaris/pull/8265/commits/f29bd0e454b56d104ebb145c85547f1a39b23429

File: .changeset/sweet-paws-tan.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-migrator': minor
```
Comment (aaronccasanova) [inline .changeset/sweet-paws-tan.md:2]:
```suggestion
'@shopify/polaris-migrator': minor
'polaris.shopify.com': minor
```

**Pull request merged**